lgc/include/lgc/patch/PatchCheckShaderCache.h

```diff
@@ -38,9 +38,35 @@ namespace lgc {
 
 // =====================================================================================================================
 // Represents the pass of LLVM patching operations for checking shader cache
-class PatchCheckShaderCache : public LegacyPatch {
+class PatchCheckShaderCache : public Patch, public llvm::PassInfoMixin<PatchCheckShaderCache> {
 public:
-  PatchCheckShaderCache();
+  // NOTE: This constructor is only used by the LegacyPatchCheckShaderCache class and can be removed once
+  // the switch to the new pass manager is completed.
+  PatchCheckShaderCache(){};
+
+  PatchCheckShaderCache(Pipeline::CheckShaderCacheFunc callbackFunc);
+
+  llvm::PreservedAnalyses run(llvm::Module &module, llvm::ModuleAnalysisManager &analysisManager);
+
+  bool runImpl(llvm::Module &module, PipelineState *pipelineState);
+
+  static llvm::StringRef name() { return "Patch LLVM for checking shader cache"; }
+
+  // Set the callback function that this pass uses to ask the front-end whether it wants to remove
+  // any shader stages. The function takes the LLVM IR module and a per-shader-stage array of input/output
+  // usage checksums, and it returns the shader stage mask with bits removed for shader stages that it wants
+  // removed.
+  void setCallbackFunction(Pipeline::CheckShaderCacheFunc callbackFunc) { m_callbackFunc = callbackFunc; }
```

Comment:
I'd suggest calling this `setCheckShaderCacheCallbackFunction` as "callback function" is not self-explanatory.